src/libcore/mem.rs

```diff
@@ -611,3 +611,80 @@ pub const POST_DROP_USIZE: usize = POST_DROP_U64 as usize;
 pub unsafe fn transmute_copy<T, U>(src: &T) -> U {
     ptr::read(src as *const T as *const U)
 }
+
+/// Returns the value of the discriminant for the enum variant in `v`.
+///
+/// The returned discriminant is cast to `u64`, no matter what the actual
+/// representation is. Note that if the enum's representation is signed, then
+/// the discriminant values returned by this function will not compare in the
+/// same order as actual enum variants would under `#[derive(PartialOrd)]`.
+///
+/// If `T` is not an enum, the return value is unspecified (this is **not** undefined
+/// behavior, but the defined behavior is that the value is unpredictable and subject
+/// to change).
+///
+/// # Stability
+///
+/// Discriminants can change if enum variants are reordered, if a new variant is added
+/// in the middle, or (in the case of a C-like enum) if explicitly set discriminants are changed.
+/// Therefore, relying on the discriminants of enums outside of your crate may be a poor decision.
+/// However, discriminants of an identical enum should not change between minor versions of the
+/// same compiler.
+///
+/// # Examples
+///
+/// Usually, the discriminant is set by the order of variants in an enum, starting at zero
+/// and counting up from there. In the case of a C-like enum, the discriminant may be set
+/// explicitly in the definition (and the result of this function is the same as casting a
+/// variant to `u64` using `as`).
+///
+/// ```
+/// #![feature(discriminant_value)]
+/// use std::mem;
+///
+/// // This enum's discriminants follow from the variant order.
+/// enum Foo {
+///     X,                      // discriminant: 0
+///     Y(i32),                 // discriminant: 1
+///     Z(Option<&'static str>) // discriminant: 2
+/// }
```

Comment:
This is leaking-documenting the “guarantees” about discriminants in sum-types again.